artifacts/definitions/Windows/Forensics/BulkExtractor.yaml

```diff
@@ -0,0 +1,163 @@
+author: Matt Green - @mgreen27
+ 
+precondition: SELECT OS From info() where OS = 'windows'
+
+parameters:
+  - name: binaryURL
+    description: |
+      Specify this as the base of the binary store (if empty we use
+      the server's public directory).
+  - name: Target
+    description: "Target. Can by physical drive, \\\\?\\HarddiskVolumeShadowCopy1 or C:\\Folder\\Path"
+    default: \\.\PhysicalDrive0
+  - name: TargetAllPhysical
+    description: "Target all attached physical drives"
+    type: bool
+  - name: TargetVSS 
+    description: "Target all VSC. Note: Not targeted to folder. Velociraptor CAN collect from the Volume Shadow direct targeted to folder with ntfs accessor so there may be a better way."
+    type: bool
+  - name: CarveEvtx
+    description: "Carve EVTX files"
+    type: bool
+  - name: FindRegex
+    description: "Regex for Bulk_extractor find plugin"
+  - name: FreeCommand
+    description: "Bulk_extractor custom commands. .e.g '-E evtx, -e zip -S unzip_carve_mode=2'"
+   
+sources:
+  - queries:
+    - LET bin <= SELECT * 
+        FROM Artifact.Generic.Utils.FetchBinary(
+            binaryURL=binaryURL, ToolName="Bulk_Extractor")
+    - LET tempfolder <= tempdir
+    - LET zip_file <= SELECT 
+            copy(filename=FullPath, dest=tempfile(extension=".exe"), 
+                accessor='zip') AS Payload 
+        FROM glob(globs="file:///" + (bin[0]).FullPath + 
+             "#/bulk_extractor.exe", accessor='zip')
+    - LET target = SELECT 
+            DeviceID,
+            if(condition=DeviceID=~"^\\\\\\\\.\\\\",
+                then=split(string=split(string=DeviceID,sep='\\\\\\\\.\\\\')[1],
+                    sep='\\\\')[0],
+                else="bulk_out") as base,
+            _DeviceID
+        FROM chain(
+            a={
+                SELECT 
+                    DeviceID,
+                    upcase(string=DeviceID) as _DeviceID
+                FROM Artifact.Windows.Sys.DiskInfo()
+                WHERE TargetAllPhysical
+            },
+            b={
+                SELECT 
+                    Target as DeviceID,
+                    upcase(string=Target) as _DeviceID
+                FROM scope()
+                WHERE Target =~ '.'
+            },
+            c={
+                SELECT 
+                    regex_replace(source=FullPath,
+                        re="GLOBALROOT\\\\Device\\\\",replace="")AS DeviceID,
+                    Data.ID AS ShadowCopyID,
+                    upcase(string=regex_replace(source=FullPath,
+                        re="GLOBALROOT\\\\Device\\\\",replace="")) as _DeviceID
+                FROM glob(globs='/*', accessor='ntfs')
+                WHERE ShadowCopyID AND TargetVSS
+                ORDER by FullPath
+            })
+            GROUP BY _DeviceID
+    - LET cmdline = SELECT (zip_file[0].Payload,'-q','99999999999','-R') + CMD + ('-o') as CMD FROM switch(
+            a= { 
+                SELECT split(string=FreeCommand, sep=' ') AS CMD 
```

Comment:
If you are allowing this you need to set required_permissions to EXECVE since it will allow users to run arbitrary commands (normally only admins are allowed). Probably this is a good idea anyway since this is an expensive artifact to collect.

https://github.com/Velocidex/velociraptor/blob/master/artifacts/proto/artifact.proto#L150

Reply:
added required permission into artifact. I think this is a good idea due to the potential heavy cpu use.